geoh5py/workspace/workspace.py

```diff
@@ -622,16 +627,16 @@ def remove_children(self, parent, children: list):
                     H5Writer.add_or_update_property_group, child, remove=True, mode="r+"
                 )
             else:
-                ref_type = self.str_from_type(child)
+                ref_type = str_from_type(child)
                 self._io_call(
                     H5Writer.remove_child, child.uid, ref_type, parent, mode="r+"
                 )
 
-    def remove_entity(self, entity: Entity | PropertyGroup):
+    def remove_entity(self, entity: Entity | PropertyGroup | EntityType):
         """
         Function to remove an entity and its children from the workspace.
         """
-        if not entity.allow_delete:
+        if hasattr(entity, "allow_delete") and not entity.allow_delete:
```

Comment:
or:


if not getattr(entity, "allow_delete", True)?

Reply:
Nice yep. I am also adding a new flag to `force` in case we are dealing with nested entities.